packages/utils/upgrade/resources/codemods/5.0.0/entity-service-document-service.code.ts

```diff
@@ -0,0 +1,368 @@
+  ...objectParam_9_1,
+};
+
+strapi.entityService.findOne(uid, ...[entityId, [objectParam_9]]);
+
+Case:  even more complex
+
+const objectParam_10_1 = {
+  publicationState: "preview",
+};
+
+const objectParam_10_2 = [uid, ...[12], ...[objectParam_10_1]];
+const objectParam_10 = [...objectParam_10_2];
+
+strapi.entityService.findOne(...[...objectParam_10]);
+
+*/
+
+const transformDeclaration = (path: ASTPath<any>, name: any, j: JSCodeshift) => {
+  const declaration = findClosesDeclaration(path, name, j);
+
+  if (!declaration) {
+    return;
+  }
+
+  transformElement(path, declaration.init, j);
+};
+
+const transformElement = (path: ASTPath<any>, element: any, j: JSCodeshift) => {
+  switch (true) {
+    case j.ObjectExpression.check(element): {
+      transformObjectParam(path, element, j);
+      break;
+    }
+
+    case j.Identifier.check(element): {
+      transformDeclaration(path, element.name, j);
+      break;
+    }
+
+    case j.SpreadElement.check(element): {
+      transformElement(path, element.argument, j);
+      break;
+    }
+
+    case j.ArrayExpression.check(element): {
+      element.elements.forEach((element) => {
+        transformElement(path, element, j);
+      });
+      break;
+    }
+    default: {
+      break;
+    }
+  }
+};
+
+const transformObjectParam = (path: ASTPath<any>, expression: ObjectExpression, j: JSCodeshift) => {
+  expression.properties.forEach((prop) => {
+    switch (true) {
+      case j.ObjectProperty.check(prop): {
+        if (!j.Identifier.check(prop.key) && !j.Literal.check(prop.key)) {
+          return;
+        }
+
+        if (j.Identifier.check(prop.key) && prop.key.name !== 'publicationState') {
+          return;
+        }
+
+        if (j.Literal.check(prop.key) && prop.key.value !== 'publicationState') {
+          return;
+        }
+
+        if (j.Identifier.check(prop.key) && prop.key.name === 'publicationState') {
+          if (!prop.computed && !prop.shorthand) {
+            prop.key.name = 'status';
+          }
+
+          if (prop.shorthand && !prop.computed) {
+            prop.shorthand = false;
+            prop.key = j.identifier('status');
+            prop.value = j.identifier('publicationState');
+          }
+        } else if (j.Literal.check(prop.key) && prop.key.value === 'publicationState') {
+          prop.key.value = 'status';
+        }
+
+        switch (true) {
+          case j.Literal.check(prop.value): {
+            prop.value = prop.value.value === 'live' ? j.literal('published') : j.literal('draft');
+
+            break;
+          }
+          case j.Identifier.check(prop.value): {
+            const declaration = findClosesDeclaration(path, prop.value.name, j);
+
+            if (!declaration) {
+              return;
+            }
+
+            if (j.Literal.check(declaration.init)) {
+              declaration.init =
+                declaration.init.value === 'live' ? j.literal('published') : j.literal('draft');
```

Comment:
Those don't mean the same thing though. We could prob add a warning about this somewhere in the doc?

Reply:
from a data perspective they mean the same thing

Reply:
Doesn't `live` means `draft` + `published` and `published` means _only_ the `published` ones?

Reply:
no

preview => published + draft data 
live => only published

now

draft => only drafts (but we always have a draft so if no changes where made it's the same data)
published: only published

Reply:
Yeah, I inverted it but basically draft means 2 things whereas live means only published.

> but we always have a draft so if no changes where made it's the same data

Not sure I understand exactly how that's possible but alright